enclave-runtime/src/lib.rs

```diff
@@ -476,31 +521,9 @@ pub unsafe extern "C" fn sync_parentchain(
 /// * validates and execute those extrinsics (containing indirect calls), mutating state
 /// * sends `confirm_call` xt's of the executed unshielding calls
 /// * sends `confirm_blocks` xt's for every synced parentchain block
-fn sync_parentchain_internal<PB>(blocks_to_sync: Vec<SignedBlockG<PB>>) -> Result<()>
-where
-	PB: BlockT<Hash = H256, Header = ParentchainHeader>,
-	NumberFor<PB>: BlockNumberOps,
-{
-	let validator_access = Arc::new(ValidatorAccessor::<PB>::default());
-	let genesis_hash = validator_access.execute_on_validator(|v| v.genesis_hash(v.num_relays()))?;
-
-	let signer = Ed25519Seal::unseal()?;
-	let shielding_key = Rsa3072Seal::unseal()?;
-	let stf_executor =
-		Arc::new(StfExecutor::new(Arc::new(OcallApi), Arc::new(GlobalFileStateHandler)));
-	let extrinsics_factory =
-		Arc::new(ExtrinsicsFactory::new(genesis_hash, signer, GLOBAL_NONCE_CACHE.clone()));
-	let indirect_calls_executor =
-		Arc::new(IndirectCallsExecutor::new(shielding_key, stf_executor.clone()));
-	let parentchain_block_importer = ParentchainBlockImporter::new(
-		validator_access,
-		Arc::new(OcallApi),
-		stf_executor,
-		extrinsics_factory,
-		indirect_calls_executor,
-	);
+fn sync_parentchain_internal(blocks_to_sync: Vec<SignedBlock>) -> Result<()> {
+	let block_import_dispatcher =
+		GLOBAL_DISPATCHER_COMPONENT.get().ok_or(Error::ComponentNotInitialized)?;
 
-	parentchain_block_importer
-		.import_parentchain_blocks(blocks_to_sync)
-		.map_err(|e| e.into())
+	block_import_dispatcher.dispatch_blocks(blocks_to_sync).map_err(|e| e.into())
```

Comment:
This has become very simple now - just get the dispatcher instance and dispatch all the blocks to be imported.